test/Microsoft.AspNetCore.HttpsEnforcement.Tests/HttpsRedirectionMiddlewareTests.cs

```diff
@@ -157,5 +178,71 @@ public class HttpsRedirectionMiddlewareTests
 
             Assert.Equal(expectedUrl, response.Headers.Location.ToString());
         }
+
+        [Fact]
+        public async Task SetServerAddressesFeature_SingleHttpsAddress_Success()
+        {
+            var builder = new WebHostBuilder()
+               .ConfigureServices(services =>
+               {
+                   services.AddHttpsRedirection(options =>
+                   {
+                   });
+               })
+               .Configure(app =>
+               {
+                   app.UseHttpsRedirection();
+                   app.Run(context =>
+                   {
+                       return context.Response.WriteAsync("Hello world");
+                   });
+               });
+
+            var featureCollection = new FeatureCollection();
+            featureCollection.Set<IServerAddressesFeature>(new ServerAddressesFeature());
+            var server = new TestServer(builder, featureCollection);
+
+            server.Features.Get<IServerAddressesFeature>().Addresses.Add("https://localhost:5050");
+            var client = server.CreateClient();
+
+            var request = new HttpRequestMessage(HttpMethod.Get, "");
+
+            var response = await client.SendAsync(request);
+
+            Assert.Equal("https://localhost:5050/", response.Headers.Location.ToString());
+        }
+
+        [Fact]
+        public async Task SetServerAddressesFeature_MultipleHttpsAddresses_ThrowInMiddleware()
```

Comment:
Add a test that has multiple HTTPS urls but all of them pointing to the same port as @Tratcher suggested